### PR TITLE
Admin notice for requirements

### LINKED ISSUE #893: Add admin notice when status page requirements are not met
If any of the status page requirements are not met, an admin notice can be displayed to inform customers about that. The notice can include a link to the status page and a dismiss button.

### PR DESCRIPTION
**Pull request by MohamadNateqi**
close #893

### CODE REVIEW

File: includes/Main.php [hunk 1]
```diff
@@ -873,6 +876,46 @@ public function no_dir_notice() {
...
+				return;
+			}
+		}
+
+		$status_page_url = admin_url( 'admin.php?page=wpo_wcpdf_options_page&tab=debug&section=status' );
+		$dismiss_url     = wp_nonce_url( add_query_arg( 'wpo_dismiss_requirements_notice', true ), 'dismiss_requirements_notice' );
+		$notice_message  = sprintf(
+			/* translators: 1: Plugin name, 2: Open anchor tag, 3: Close anchor tag */
+			__( 'Your server does not meet the requirements for %1$s. Please check the %2$sStatus page%3$s for more information.', 'woocommerce-pdf-invoices-packing-slips' ),
+			'<strong>WooCommerce PDF Invoices & Packing Slips</strong>',
```
Comment (alexmigf) [inline includes/Main.php:904]: This name is wrong.

**Review by alexmigf — CHANGES_REQUESTED** [review]
GMagick or IMagick is only recommended, but I still have the notice showing:

Screenshot from 2024-11-21 08-36-08

**Review by alexmigf — COMMENTED** [review]
There's an issue with this implementation: the notice still appears even after meeting the requirements, as long as you don't visit the status page. I believe we should refactor the logic to separate it from the view.

File: includes/Settings/SettingsDebug.php [hunk 2]
```diff
@@ -924,6 +926,209 @@ public function document_link_access_type_table() {
...
+			} elseif ( $apc ) {
+				$server_configs['opcache']['value'] = 'APC ' . phpversion( 'apc' );
+			} elseif ( $zop ) {
+				$server_configs['opcache']['value'] = 'Zend OPCache ' . phpversion( 'Zend OPcache' );
+			} else {
+				$server_configs['opcache']['value'] = 'PHP OPCache ' . phpversion( 'opcache' );
+			}
+		}
+
+		if ( ! $server_configs['PHP version']['result'] ) {
```
Comment (alexmigf) [inline includes/Settings/SettingsDebug.php:1056]: I believe this section can be removed, as promoting PHP 5 is no longer necessary or recommended.

File: views/advanced-status.php [hunk 3]
```diff
@@ -3,143 +3,7 @@
...
-	}
-
-	if ( ! $config['result'] ) {
-		$show_requirement_notice = true;
-		break;
-	}
-}
-
-update_option( 'wpo_wcpdf_show_requirement_notice', $show_requirement_notice );
+$server_configs = $this->get_server_config();
```
Comment (alexmigf) [inline views/advanced-status.php:6]: Maybe you could pass this variable from `display_status()` or `output()`.

**Review by alexmigf — CHANGES_REQUESTED** [review]
This branch doesn't include the submodule. Perhaps you could copy your changes to a new one? I'm unsure if merging with the main branch would be seamless.

**Comment by MohamadNateqi** [comment]
@alexmigf It seems the submodule is included in this branch, can you also check?

**Review by alexmigf — APPROVED** [review]
Good job!

**Pull request merged**